src/transformers/keras_callbacks.py

```diff
@@ -0,0 +1,94 @@
+from pathlib import Path
+from time import sleep
+from typing import Optional, Union
+
+from tensorflow.keras.callbacks import Callback
+
+from huggingface_hub import Repository
+
+from . import IntervalStrategy, PreTrainedTokenizerBase
+from .file_utils import get_full_repo_name
+
+
+class PushToHubCallback(Callback):
+    def __init__(
+        self,
+        output_dir: Union[str, Path],
+        save_strategy: Union[str, IntervalStrategy] = "epoch",
+        save_steps: Optional[int] = None,
+        tokenizer: Optional[PreTrainedTokenizerBase] = None,
+        hub_model_id: Optional[str] = None,
+        hub_token: Optional[str] = None,
+    ):
+        """
+        output_dir (:obj:`str`):
+            The output directory where the model predictions and checkpoints will be written. The name of the
+            repository to keep in sync with the local `output_dir`. Should be the whole repository
```

Comment:
I had a hard time understanding that docstring as it gives what seems to be unrelated information.

Reply:
Agreed, I copy-pasted it from elsewhere but it feels a bit incomplete - I might have missed something or mangled it somehow. I rewrote it.